src/mcts/search.cc

```diff
@@ -786,6 +786,7 @@ void SearchWorker::FetchNNResults() {
     node->SetV(-computation_->GetQVal(idx_in_computation));
     // Populate P values.
     float total = 0.0;
+    int total_children = 0;
```

Comment:
node is guaranteed to have children here, so no need to count them.
Instead `assert(node->HasChildren());` would be useful.

`force_noise` may be unneeded too, `total == 0.0f` (or even calling ApplyDirichletNoise() right after filling the vector with values) may work, although I'm not sure.

..and actually, no need to initialize it. Passing 1 as `eps` param to ApplyDirichletNoise() probably does exactly the same. Or maybe not exactly, but still good enough for sure.

Reply:
total children is being counted so we can create a correct size uniform distribution - the fact that I use it in the if statement is just defensive programming.

We could use eps of 1 instead of counting - it will make the distribution even more random though.

Reply:
Reason why I'm not advocating for more random is because the only reason I want any random at all is to ensure that the policy priors are not all *exactly* equal.
That being said I can't justify why more random would necessarily be bad.

Reply:
Happy to make changes here - but is there a decision about what to?